zio-http/src/main/scala/zhttp/socket/WebSocketFrame.scala

```diff
@@ -5,75 +5,40 @@ import io.netty.handler.codec.http.websocketx.{WebSocketFrame => JWebSocketFrame
 import zio.Chunk
 
 sealed trait WebSocketFrame extends Product with Serializable { self =>
+  def isFinal: Boolean = true
+
   final def toWebSocketFrame: JWebSocketFrame = WebSocketFrame.toJFrame(self)
-  def isFinal: Boolean                        = true
 }
 
 object WebSocketFrame {
 
-  case class Binary(bytes: Chunk[Byte]) extends WebSocketFrame { override val isFinal: Boolean = true }
-  object Binary {
-    def apply(bytes: Chunk[Byte], isFinal: Boolean): Binary        = {
-      val arg = isFinal
-      new Binary(bytes) { override val isFinal: Boolean = arg }
-    }
-    def unapply(frame: WebSocketFrame.Binary): Option[Chunk[Byte]] = Some(frame.bytes)
-  }
-
-  case class Text(text: String) extends WebSocketFrame { override val isFinal: Boolean = true }
-  object Text {
-    def apply(text: String, isFinal: Boolean): Text         = {
-      val arg = isFinal
-      new Text(text) { override val isFinal: Boolean = arg }
-    }
-    def unapply(frame: WebSocketFrame.Text): Option[String] = Some(frame.text)
-  }
-
-  final case class Close(status: Int, reason: Option[String]) extends WebSocketFrame
-
-  case object Ping extends WebSocketFrame
-  case object Pong extends WebSocketFrame
-
-  case class Continuation(buffer: ByteBuf) extends WebSocketFrame { override val isFinal: Boolean = true }
-  object Continuation {
-    def apply(buffer: ByteBuf, isFinal: Boolean): Continuation       = {
-      val arg = isFinal
-      new Continuation(buffer) { override val isFinal: Boolean = arg }
+  private[zhttp] def unsafeFromJFrame(jFrame: JWebSocketFrame): WebSocketFrame =
+    jFrame match {
+      case _: PingWebSocketFrame   => Ping
+      case _: PongWebSocketFrame   => Pong
+      case m: BinaryWebSocketFrame => Binary(Chunk.fromArray(ByteBufUtil.getBytes(m.content())), m.isFinalFragment)
+      case m: TextWebSocketFrame   => Text(m.text(), m.isFinalFragment)
+      case m: CloseWebSocketFrame  => Close(m.statusCode(), Option(m.reasonText()))
+      case m: ContinuationWebSocketFrame => Continuation(m.content(), m.isFinalFragment)
+      case _                             => null
     }
```

Comment:
Note: Only change is `unsafeFromJFrame` been added. Rest is just re-arranging of methods.